### DISCUSSION TITLE
在addEffects接口中使用onFormReact钩子不生效？

### DISCUSSION THREAD

**Discussion by MisicDemone** [alibaba/formily#2067] [Q&A 2.x]
demo可见：https://codesandbox.io/s/sharp-ives-88p0n?file=/App.tsx:315-326

由于addEffects时已经在init生命周期之后了，所以其中的依赖并没有被收集？

目前需要依赖onFormReact的依赖收集能力来实现一些视图交互（表格筛选表单的展开收起），想把这部分逻辑收到一个自定义组件里面，不能再form创建的时候传入effects。

**Comment by janryWang** [comment]
如果是为了依赖收集，observer内部本身就能收集依赖

**Reply by MisicDemone** [reply]
确实可以直接写autorun。不过onFormReact以及onFormInit这些hook没法在addEffects中生效这个没能在文档里体现出来，还得看源码才能知道。

**Reply by janryWang** [reply]
addEffects其实就是注册事件，初始化已经执行过了是肯定不会再次执行的
